stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1280,9 +1276,10 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     public mutating func append<SourceType>(_ buffer : UnsafeBufferPointer<SourceType>) {
         if buffer.count == 0 { return }
         if !isKnownUniquelyReferenced(&_backing) {
-            _backing = _backing.mutableCopy()
+            _backing = _backing.mutableCopy(_sliceRange)
```

Comment:
Not something that needs to happen now, but there's an opportunity to make this more efficient by allocating the new backing at the right size initially, rather than copy + resize.